bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/product/ProductImplTest.java

```diff
@@ -360,6 +360,12 @@ public void testGroupedProduct() throws IOException {
             Assert.assertEquals(pi.getName(), item.getName());
             Assert.assertEquals(pi.getPriceRange().getMinimumPrice().getFinalPrice().getValue(), item.getPriceRange().getFinalPrice(), 0);
             Assert.assertEquals(gp.getItems().get(i).getQty(), item.getDefaultQuantity(), 0);
+
+            if (i < 3) {
```

Comment:
To avoid the check based on the position, you could do 
`Assert.assertEquals(pi.get("__typename").equals("VirtualProduct"), item.isVirtualProduct());`

Reply:
Or `Assert.assertEquals(pi instanceof VirtualProduct, item.isVirtualProduct());`